### ISSUE TITLE
Disallow namespace objects in JSX

### ISSUE THREAD

**Issue by nzakas** [eslint/espree#261]
We were parsing the following invalid JSX without error:

```jsx
<a.b:c></a.b:c>;
```

This was due to an issue in Acorn-JSX that has now been resolved: https://github.com/RReverser/acorn-jsx/issues/27

We should update to ensure that we no longer parse this syntax.

**Comment by RReverser** [comment]
Just curious - wouldn't you want to prohibit namespaces in general? (as they are very uncommon and not used by any of the popular transformers)

**Comment by nzakas** [comment]
While not common, that are part of the spec. I'd rather be spec-compliant and not have to worry about what comes in and out of favor going forward.

**Comment by kaicataldo** [comment]
I can take this on, but read through the acorn-jsx thread in the original post and wasn't quite sure what the intended change is. According to [this comment](https://github.com/RReverser/acorn-jsx/issues/27#issuecomment-146983354), it sounds like we shouldn't be able to mix dots and semicolons. Is that right? Sorry, not very familiar with JSX.

**Comment by RReverser** [comment]
@kaicataldo That's right.

**Comment by kaicataldo** [comment]
Awesome, thanks - PR opened
